src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -4,15 +4,31 @@
 #---------------------------------------------------------------------------------------------
 #pylint: disable=line-too-long
 
-from azure.cli.core.commands.parameters import (location_type, get_enum_choices)
+from azure.cli.core.commands.parameters import (location_type, get_enum_choices, get_resource_name_completion_list)
 from azure.cli.core.commands import register_cli_argument
-
 from azure.mgmt.iothub.models.iot_hub_client_enums import IotHubSku
+from ._factory import iot_hub_service_factory
+from .custom import iot_device_list
+
+
+def get_device_id_completion_list():
+    def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-argument
+        client = iot_hub_service_factory(kwargs)
+        if parsed_args.hub_name:
+            return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)]
+        else:
+            return []
+    return completer
 
 
+register_cli_argument('iot', 'hub_name', options_list=('--hub-name', '--name', '--hub', '-n'),
```

Comment:
This is to take suggestion from Travis. Read more details at #953 .
Also there is consideration for backward compatibility. We have released document (preview) referencing these commands. We will remove extra parameters after the document is updated.